kore/src/Kore/Step/Axiom/Matcher.hs

```diff
@@ -668,31 +678,63 @@ matchNormalizedAc
     -> Builtin.NormalizedAc normalized (TermLike Concrete) (TermLike variable)
     -> Builtin.NormalizedAc normalized (TermLike Concrete) (TermLike variable)
     -> MaybeT (MatcherT variable unifier) ()
-matchNormalizedAc pushValue wrapTermLike normalized1 normalized2
-  | [] <- excessAbstract1
-  = do
-    Monad.guard (null excessConcrete1)
-    case opaque1 of
-        [] -> do
-            Monad.guard (null opaque2)
-            Monad.guard (null excessConcrete2)
-            Monad.guard (null excessAbstract2)
-        [frame1]
-          | null excessAbstract2
-          , null excessConcrete2
-          , [frame2] <- opaque2 ->
-            push (Pair frame1 frame2)
-          | otherwise ->
-            push (Pair frame1 normalized2')
-        _ -> empty
-    lift $ Foldable.traverse_ pushValue concrete12
-    lift $ Foldable.traverse_ pushValue abstractMerge
+matchNormalizedAc pushElement pushValue wrapTermLike normalized1 normalized2
+    | [element1] <- abstract1
+    , [frame1] <- opaque1
+    , null concrete1 = do
+        let (key1, value1) = Builtin.unwrapElement element1
+        case Builtin.lookupSymbolicKeyOfAc key1 normalized2 of
+            Just value2 -> do
+                lift $ pushValue (Pair value1 value2)
```

Comment:
It would be better to move `lift` to the start of the block, that is `lift $ do ...`